src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -162,6 +164,13 @@ private void ConnectToWorkspace(Workspace workspace)
 
             public void DisconnectFromWorkspace()
             {
+                if (_reportChangeCancellationSource != null)
+                {
+                    _reportChangeCancellationSource.Cancel();
+                    _reportChangeCancellationSource.Dispose();
+                    _reportChangeCancellationSource = null;
```

Comment:
I suspect this is now racing against EnqueueParseSnapshotWorkerAsync where this could dispose and null out the field but EnqueueParseSnapshotWorkerAsync will then null ref.

Reply:
➡️ I rewrote the first commit of this pull request to use a completely new approach for this class.